### PR TITLE
chore(OKR): Lint configuration synced with master

### PR DESCRIPTION
**Pull request by mkhizeryounas**
### Changelog
- Removed following ignored rules by updating the code
  - `max-len`
  - `implicit-arrow-linebreak`
  - `@typescript-eslint/no-explicit-any`
- Removed unused rules patterns from `import/no-extraneous-dependencies` as those are overridden by `TS` configuration
- Added `import/no-extraneous-dependencies` rule for `stories` in overrides

### CODE REVIEW

File: .eslintrc
Comment (tecnoloco) [inline .eslintrc:76]: > Removed following ignored rules by updating the code
> max-len
> implicit-arrow-linebreak
> @typescript-eslint/no-explicit-any

Does this mean all arrow line breaks errors are resolved?

Comment (mkhizeryounas) [reply .eslintrc:76]: Yes

File: .eslintrc
Comment (mkhizeryounas) [reply .eslintrc:71]: These rules are overridden by TS config so added this rule just to make sure the conventions followed in the project is satisfied.

**Review by tecnoloco — APPROVED** [review]
No further comments from my side. LGTM. I would recommend another pair of eyes before merging though.

File: src/services/navigator/helpers.test.ts
Comment (flochtililoch) [inline src/services/navigator/helpers.test.ts:5]: is this change correct? It doesn't look like this PR renames the file.

File: test/helpers.tsx
Comment (flochtililoch) [inline test/helpers.tsx:31]: curious why this is needed

File: .eslintrc
Comment (flochtililoch) [inline .eslintrc:53]: any reasons to remove the two other extended plugins?

File: .flowconfig
Comment (mkhizeryounas) [reply .flowconfig:17]: yes, otherwise flow is failing.

File: src/services/dom/types.ts
Comment (mkhizeryounas) [reply src/services/dom/types.ts:37]: this is because we have a limit of 100 chars but this becomes bigger than 100 chars so to fix that changed this here.

File: src/services/navigator/helpers.test.ts
Comment (mkhizeryounas) [reply src/services/navigator/helpers.test.ts:5]: yes, due to this tests were failing as this file doesn't exist

File: test/helpers.tsx
Comment (mkhizeryounas) [reply test/helpers.tsx:31]: lint fails, complaining that unexpected arrow function.

File: .eslintrc
Comment (mkhizeryounas) [reply .eslintrc:53]: Not really, these didn't exist in master so just to sync both.

File: src/services/inline-context/index.ts
Comment (mkhizeryounas) [reply src/services/inline-context/index.ts:53]: Max len rule fails if we don't do this.

**Review by flochtililoch — APPROVED** [review]

**Pull request merged**
